Generator/src/main/java/com/microsoft/fluentui/generator/model/TypographyGenerator.kt

```diff
@@ -0,0 +1,167 @@
+                                                    )
+                                                )
+                                            }
+                                        }
+                                    }
+                                }
+                                else -> Unit
+                            }
+                        }
+                    }
+                    is String ->
+                        if (Regex(FONT_PATTERN).matches(value.toString())) {
+                            val stylesKeys = mutableMapOf<StyleKeys, String>()
+                            stylesKeys[StyleKeys.TEXT_SIZE] = "@dimen/font_size_${entry.key}"
+                            stylesKeys[StyleKeys.FONT_FAMILY] =
+                                "\"${TextStyles.getStyleByName(entry.key)?.fontFamily}\""
+                            stylesKeys[StyleKeys.FONT_WEIGHT] = "${TextStyles.getStyleByName(entry.key)?.fontWeight}"
+                            stylesList.add(
+                                Style(
+                                    "${name}_${entry.key}".decapitalize(),
+                                    null,
+                                    stylesKeys,
+                                    false
+                                )
+                            )
+                        }
+                }
+            }
+            if (stylesList.isNotEmpty()) {
+                generateStyles(name, stylesList)
+                generateStyles(name,stylesList,true)
+            }
+        }
+    }
+
+    private fun generateSizeResources(key: String, value: ArrayList<*>) {
+        value.forEach {
+            when (it) {
+                is LinkedHashMap<*, *> -> {
+                    val entry = it.values.single().toString()
+                    if (Regex(FONT_PATTERN).matches(entry)) {
+                        val resKey = it.keys.single().toString()
+                        "\\d{1,2}".toRegex().find(entry)?.let { fontDefinition ->
+                            addResource("font_size_${key}_$resKey", fontDefinition.value, ResourceType.TYPOGRAPHY)
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private fun generateTextStyleSizeResources(value: ArrayList<*>) {
+        value.forEach { fontDefinition ->
+            if (fontDefinition is LinkedHashMap<*, *>) {
+                fontDefinition.forEach {
+                    val key = it.key
+                    val textStyle = TextStyles.getStyleByName(it.key.toString())
+                    textStyle?.let { style ->
+                        addResource(
+                            "font_size_$key",
+                            style.fontSize.toString(),
+                            ResourceType.TYPOGRAPHY
+                        )
+                    }
+                }
+            }
+        }
+    }
+
+    // FIXME: Fix font weights
```

Comment:
yes we do use fontweights now.
It is to be fixed only for one case now when we explicitly define font in stylesheet without consuming them directly.
Will be fixed fully when we consume Avatar view
Here is an example: - 
textFont: [ xSmall: "Font(9px, regular)"]
It will be converted to something like 
textFont: [ xSmall: "Font(400,9px, "san-serif")"]